pkg/guestagent/kubernetesservice/kubernetesservice.go

```diff
@@ -0,0 +1,130 @@
+package kubernetesservice
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"net/url"
+	"os"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+type Protocol string
+
+const (
+	// UDP/SCTP when lima port forwarding works on those protocols
+	TCP Protocol = "TCP"
+)
+
+type Entry struct {
+	Protocol Protocol
+	IP       net.IP
+	Port     uint16
+}
```

Comment:
I feel the need of unifying port detecting packages(prochttp, iptables, kubernetesservice) if this PR is merged. These packages should use the same `Entry` struct for instance.